common/openapi.yml

```diff
@@ -297,6 +321,36 @@ components:
       description: >
         DatasetPreviewTablesResponse returns information about the table structure of a dataset.
 
+    DatasetPreviewTable:
+      type: object
+      properties:
+        columns:
+          type: array
+          items:
+            $ref: '#/components/schemas/ColumnModel'
+        rows:
+          type: array
+          items:
+            type: object
+
+    ColumnModel:
+      type: object
+      properties:
+        name:
+          type: string
+        datatype:
+          $ref: '#/components/schemas/TableDataType'
+        array_of:
+          type: boolean
+          description: if true, make this column an array of type datatype.
+          default: false
+
+    TableDataType:
+      type: string
+      enum: [ string, boolean, bytes, date, datetime, dirref, fileref, float, float64, integer, int64, numeric, record, text, time, timestamp ]
```

Comment:
I think I mentioned this before, but I'm not sure about the use of including column types in the response. The UI doesn't currently use types, and we don't know what types other storage systems support. When we add new systems, we'll have to map that systems's types to our types. I think it would be better to only include data the UI needs to limit the the scope of our API, and to lower the cost of adding new storage systems.